api/middleware/authorization.go

```diff
@@ -6,56 +6,51 @@ import (
 	"net/http"
 	"strings"
 
+	"golang.org/x/exp/slices"
+
 	"github.com/caraml-dev/mlp/api/pkg/authz/enforcer"
 )
 
+type Authorizer struct {
+	authEnforcer enforcer.Enforcer
+}
+
 func NewAuthorizer(enforcer enforcer.Enforcer) *Authorizer {
 	return &Authorizer{authEnforcer: enforcer}
 }
 
-type Authorizer struct {
-	authEnforcer enforcer.Enforcer
+type Operation struct {
+	RequestPath   string
+	RequestMethod []string
 }
 
-var methodMapping = map[string]string{
-	http.MethodGet:     enforcer.ActionRead,
-	http.MethodHead:    enforcer.ActionRead,
-	http.MethodPost:    enforcer.ActionCreate,
-	http.MethodPut:     enforcer.ActionUpdate,
-	http.MethodPatch:   enforcer.ActionUpdate,
-	http.MethodDelete:  enforcer.ActionDelete,
-	http.MethodConnect: enforcer.ActionRead,
-	http.MethodOptions: enforcer.ActionRead,
-	http.MethodTrace:   enforcer.ActionRead,
+var publicOperations = []Operation{
+	{"/projects", []string{http.MethodGet, http.MethodPost}},
```

Comment:
It's possible, but there are several downside. For example, we can define a permission called "mlp.publicresource.read", and associate the endpoints to this permission in Keto. But that means:
1. Every time we add a new endpoint we need to remember to add this resource to the Keto bootstrap, similar to the status quo.
2. We have to assign a default role to every user, that has access to this public resource read permission. And since the users sign up via Google rather than managed by MLP, that means the only time we have the opportunity to create a default role is when the user creates the project.
3. So that means, until the user create his first MLP projects, he is not able to view any of the public resources - which is different from the current behaviour.
4. Similarly, we have to explicitly make POST /projects public, rather than relying on Keto, otherwise nobody will be able to access the resource.

Reply:
As for the root cause why this is a problem with the new Keto and not old Keto: new Keto does not match via Glob. That means, "*" is not a valid option. We can't really do something like, allow resource A to be access by "*" users.

Reply:
How are the public methods expected to be handled by Oauthkeeper?

Reply:
Similar to what is implemented here in the middleware. The rule/config will look like this:
```json
apiVersion: oathkeeper.ory.sh/v1alpha1
kind: Rule
metadata:
  name: mlp-get-and-create-projects
spec:
  match:
    url: <.+>/projects
    methods:
      - "GET"
      - "POST"
  authenticators:
    - handler: jwt
  authorizer:
    handler: allow
```

Reply:
So, yes, even when we use oathkeeper, we still need to add additional rule when new endpoints are added, even though we are not adding anything new to Keto. Similar to how the middleware needs to be modified to handle new endpoints in this MR, if we are not using oathkeeper.

Reply:
Got it. Then the middleware's implementation is similar to the Oauthkeeper rules. Given that we will eventually remove this middleware, let's keep it this way.